Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Sandboxed/SelectorsPollerScript.js

```diff
@@ -114,217 +270,154 @@ window.__firefox__.execute(function($) {
-        }
-      } else if (aMutation.addedNodes.length > 0) {
-        for (let _d = 0, _e = aMutation.addedNodes; _d < _e.length; _d++) {
-          const node = _e[_d]
-          const element = asElement(node)
-          if (!element) {
-            continue
-          }
-          mutationScore++
-          const id = element.id
-          if (id && !queriedIds.has(id)) {
-            notYetQueriedIds.push(id)
-            queriedIds.add(id)
+    for (const mutation of mutations) {
+      const changedElm = mutation.target
+
+      switch (mutation.type) {
+        case 'attributes':
+          // Since we're filtering for attribute modifications, we can be certain
+          // that the targets are always HTMLElements, and never TextNode.
+          switch (mutation.attributeName) {
+            case 'class':
+              mutationScore += changedElm.classList.length
+              extractClassSelectorsIfNeeded(changedElm)
+              break
+            case 'id':
+              mutationScore++
+              extractIDSelectorIfNeeded(changedElm)
+              break
           }
-          const classList = element.classList
-          if (classList) {
-            mutationScore += classList.length
-            for (let _f = 0, _g = classList; _f < _g.length; _f++) {
-              const className = _g[_f]
-              if (className && !queriedClasses.has(className)) {
-                notYetQueriedClasses.push(className)
-                queriedClasses.add(className)
-              }
+          break
+        case 'childList':
+          if (mutation.addedNodes.length > 0) {
+            mutationScore += mutation.addedNodes.length
+
+            for (const node of mutation.addedNodes) {
+              if (!isElement(node)) { continue }
+              querySelectorsFromElement(node)
             }
           }
-        }
       }
     }
-    
+
     return mutationScore
   }
-  
-  const onMutations = (mutations, observer) => {
-    const mutationScore = queueAttrsFromMutations(mutations)
-    // Check the conditions to switch to the alternative strategy
-    // to get selectors.
-    if (CC.switchToSelectorsPollingThreshold !== undefined) {
-      const now = performance.now()
-      if (now > currentMutationStartTime + scoreCalcIntervalMs) {
-        // Start the next time frame.
-        currentMutationStartTime = now
-        currentMutationScore = 0
-      }
-      currentMutationScore += mutationScore
-      if (currentMutationScore > CC.switchToSelectorsPollingThreshold) {
-        usePolling(observer)
+
+  /**
+   * Tries to extract origin from the given URL string
+   * @param {string} urlString The string to extract the origin from
+   * @returns Origin string if found otherwise undefined
+   */
+  const extractOriginFromURLString = (urlString) => {
+    if (urlString.startsWith('//')) {
+      try {
+        const newURLString = [window.location.protocol, urlString].join('')
+        const url = new URL(newURLString)
```

Comment:
Better to write as 

`const url = new URL(urlString, window.location.toString())`

Reply:
Done